src/main/resources/META-INF/plugin.xml

```diff
@@ -367,6 +367,25 @@ L
                 id="LSPFormattingAndRangeBothService"
                 implementation="com.redhat.devtools.lsp4ij.features.formatting.LSPFormattingAndRangeBothService"/>
 
+        <!-- These provide an improved editor experience in abstract file type and TextMate files -->
```

Comment:
EP registrations for these features. Note that they are explicitly ordered to be evaluated first before any corresponding EP implementation(s).